monitoring/monitoring/utils.py

```diff
@@ -0,0 +1,218 @@
+        values="products", index=["date"], columns="type", aggfunc=np.sum, fill_value=0
+    ).sort_values(by="date", ascending=False)
+    fig_timestamps_agg = px.line(df, x="date", y="products", color="type")
+    return fig_timestamps_agg, df_timestamps_agg
+
+
+def products_by_label() -> tuple:
+    """
+    Gets 'products_by_sustainability_label' from green-db table, counts products with
+    "certificate:UNKNOWN" and products with any other label and creates a dataframe and a line
+    chart from it. It also filters given query to present products by sustainability label(s)
+    from last product extraction in a dataframe.
+
+    Returns:
+        List: [plotly chart, pandas dataframe, pandas dataframe]
+    """
+    query = green_db.products_by_sustainability_label()
+    unknown = []
+    known = []
+    for row in query:
+        for item in row:
+            if type(item) is list:
+                for label in item:
+                    if label == "certificate:UNKNOWN":
+                        unknown.append(row)
+                    else:
+                        known.append(row)
+    unknown_df = pd.DataFrame(unknown, columns=["timestamp", "labels", "count"])
+    unknown_df["date"] = pd.to_datetime(unknown_df["timestamp"]).dt.date
+    unknown_cumm = unknown_df.groupby("date").sum()
+    unknown_cumm["label"] = "certificate:UNKNOWN"
+    known_df = pd.DataFrame(known, columns=["timestamp", "labels", "count"])
+    known_df["date"] = pd.to_datetime(known_df["timestamp"]).dt.date
+    known_cumm = known_df.groupby("date").sum()
+    known_cumm["label"] = "Known certificates"
+    join_df = pd.concat([unknown_cumm, known_cumm]).reset_index()
+    fig_label_unknown = px.line(join_df, x="date", y="count", color="label")
+    df_label_unknown = join_df.pivot_table(
+        values="count", index=["date"], columns="label", aggfunc=np.sum, fill_value=0
+    ).sort_values(by="date", ascending=False)
+    all = pd.DataFrame(query, columns=["timestamp", "labels", "count"])
+    last_extraction = all[(all["timestamp"] >= green_db.get_latest_timestamp())].sort_values(
+        "count", ascending=False
+    )
+    return fig_label_unknown, df_label_unknown, last_extraction[["labels", "count"]]
+
+
+def products_unknown_label() -> Any:
+    """
+    Gets 'products_by_sustainability_label_timestamp' for last timestamp found in green-db table
+    and filter products with "certificate:UNKNOWN".
+
+    Returns:
+        Dataframe: listing product id, name, merchant, url and sustainability label(s)
+    """
+    query = green_db.products_by_sustainability_label_timestamp(green_db.get_latest_timestamp())
+    unknown = []
+    for row in query:
+        for item in row:
+            if type(item) is list:
+                for label in item:
+                    if label == "certificate:UNKNOWN":
+                        unknown.append(row)
+    return pd.DataFrame(unknown, columns=["id", "timestamp", "merchant", "name", "url", "labels"])
```

Comment:
as above

Reply:
Removed and filtered directly on the query to db ([6800091](https://github.com/calgo-lab/green-db/pull/91/commits/68000913abc6ced9cff53a0be8aa7df59596e658))